Source/JavaScriptCore/interpreter/Interpreter.cpp

```diff
@@ -516,6 +597,25 @@ void Interpreter::getStackTrace(JSCell* owner, Vector<StackFrame>& results, size
             return IterationStatus::Continue;
         }
 
+        if (Options::useAsyncStackTrace()) [[unlikely]] {
+            if (asyncFunctionResumeFunction && visitor->callee().isCell()) {
+                if (JSFunction* callee = jsDynamicCast<JSFunction*>(visitor->callee().asCell())) {
+                    if (callee == asyncFunctionResumeFunction && visitor->argumentCountIncludingThis() > 1) {
+                        // The JSGenerator argument position varies based on optimization.
+                        // Search through all arguments until we find a JSGenerator.
```

Comment:
That doesn't sound right. I think this is masking a bug. Can you investigate what is happening?